### PR TITLE
add README of object_merger

### PR DESCRIPTION
**Pull request by yukke42**
## Description

add README of object_merger

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: perception/object_recognition/detection/object_merger/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# object_merger
+
+## Purpose
+
+object_merger is a package for merging detected obejcts from two methods by data association.
```
Comment (yukkysaito) [inline perception/object_recognition/detection/object_merger/README.md:5]:
```suggestion
object_merger is a package for merging detected objects from two methods by data association.
```

File: perception/object_recognition/detection/object_merger/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## Inner-workings / Algorithms
+
+The successive shortest path algorithm is used to solve the data association problem (the minimum-cost flow problem). The cost is calculated by the distance between two objects and gate functions are applied to reset cost, s.t. the maximun distance, the maximum area and the minimum area.
```
Comment (yukkysaito) [inline perception/object_recognition/detection/object_merger/README.md:9]:
```suggestion
The successive shortest path algorithm is used to solve the data association problem (the minimum-cost flow problem). The cost is calculated by the distance between two objects and gate functions are applied to reset cost, s.t. the maximum distance, the maximum area and the minimum area.
```

File: perception/object_recognition/detection/object_merger/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+<!-- Write links you referred to when you implemented.
+
+Example:
+  [1] {link_to_a_thesis}
+  [2] {link_to_an_issue}
+-->
+
+## (Optional) Future extensions / Unimplemented parts
+
+Data assiciation algorithm was the same as that of multi_object_traker, but the algorithm of multi_object_traker was already updated.
```
Comment (yukkysaito) [inline perception/object_recognition/detection/object_merger/README.md:72]:
```suggestion
Data association algorithm was the same as that of multi_object_tracker, but the algorithm of multi_object_tracker was already updated.
```

Comment (wep21) [inline perception/object_recognition/detection/object_merger/README.md:72]:
```suggestion
Data association algorithm was the same as that of multi_object_tracker, but the algorithm of multi_object_tracker was already updated.
```

Comment (yukke42) [reply perception/object_recognition/detection/object_merger/README.md]: fixed at [3a47674](https://github.com/tier4/AutowareArchitectureProposal.iv/pull/621/commits/3a47674bde64508f0e109441d5eac12b36bd8b37)

File: perception/object_recognition/detection/object_merger/README.md (hunk 2 above)
Comment (yukke42) [reply perception/object_recognition/detection/object_merger/README.md]: fixed at [3a47674](https://github.com/tier4/AutowareArchitectureProposal.iv/pull/621/commits/3a47674bde64508f0e109441d5eac12b36bd8b37)

File: perception/object_recognition/detection/object_merger/README.md (hunk 1 above)
Comment (yukke42) [reply perception/object_recognition/detection/object_merger/README.md]: fixed at [3a47674](https://github.com/tier4/AutowareArchitectureProposal.iv/pull/621/commits/3a47674bde64508f0e109441d5eac12b36bd8b37)

**Review by yukkysaito — APPROVED** [review]

**Pull request merged**